addons/product/static/src/js/product_pricelist_report.js

```diff
@@ -1,317 +1,228 @@
-var AbstractAction = require('web.AbstractAction');
-var core = require('web.core');
-var FieldMany2One = require('web.relational_fields').FieldMany2One;
-var StandaloneFieldManagerMixin = require('web.StandaloneFieldManagerMixin');
-var Widget = require('web.Widget');
-
-var QWeb = core.qweb;
-var _t = core._t;
-
-var QtyTagWidget = Widget.extend({
-    template: 'product.report_pricelist_qty',
-    events: {
-        'click .o_remove_qty': '_onClickRemoveQty',
-    },
-    /**
-     * @override
-     */
-    init: function (parent, defaulQuantities) {
-        this._super.apply(this, arguments);
-        this.quantities = defaulQuantities;
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { _lt } from "@web/core/l10n/translation";
+import { useService } from "@web/core/utils/hooks";
+import { useSetupAction } from "@web/webclient/actions/action_hook";
+
+const { Component, useState, onWillStart, markup, useExternalListener } = owl;
+
+class GeneratePriceListReport extends Component {
+    setup() {
+        super.setup();
+
+        this.orm = useService("orm");
         this.MAX_QTY = 5;
-    },
-
-    //--------------------------------------------------------------------------
-    // Handlers
-    //--------------------------------------------------------------------------
-
-    /**
-     * Add a quantity when add(+) button clicked.
-     *
-     * @private
-     */
-    _onClickAddQty: function () {
-        if (this.quantities.length >= this.MAX_QTY) {
-            this.displayNotification({ message: _.str.sprintf(
-                _t("At most %d quantities can be displayed simultaneously. Remove a selected quantity to add others."),
-                this.MAX_QTY
-            ) });
+        this.action = useService("action");
+        this.active_model = this.props.action.context.active_model;
+        this.active_ids = this.props.action.context.active_ids;
+
+        let pastState = this.props.state || Object;
+
+        this.state = useState({
+            quantities: pastState.quantities || [1, 5, 10],
+            displayPricelist: pastState.displayPricelist || false,
+            pricelists: [],
+            selectedPricelist: pastState.selectedPricelist || {},
+            html: markup(""),
+        });
+
+        onWillStart(async () => {
+            this.state.pricelists = await this.getPricelist()
+            this.state.selectedPricelist = this.state.selectedPricelist || this.state.pricelists[0];
+
+            this.renderHtml();
+       });
+
+       useSetupAction({
+            getLocalState: () => {
+                return {
+                    displayPricelist: this.state.displayPricelist,
+                    selectedPricelist: this.state.selectedPricelist,
+                    quantities: this.state.quantities,
+                };
+            },
+        });
+
+       useExternalListener(window, "click", this.onClickLink);
+    }
+
+    getTitle() {
+        return _lt("Pricelist Report")
```

Comment:
During the life of your component you have to use approx._t. _lt is when you have no env.